library/src/scala/tasty/reflect/SourceCodePrinter.scala

```diff
@@ -1179,7 +1179,7 @@ class SourceCodePrinter[R <: Reflection & Singleton](val tasty: R)(syntaxHighlig
         printType(body)
 
       case ParamRef(lambda, idx) =>
-        lambda match {
+        (lambda: @unchecked) match {
```

Comment:
@liufengyun I was getting the following exhaustivity fatal warning where `lambda: MethodType | PolyType | TypeLambda`. Same if did the cases with `case lambda: MethodType => ...`. It might be related to having a class tag preforming the type test.
```scala
[error] 1182 |        lambda match {
[error]      |        ^^^^^^
[error]      |match may not be exhaustive.
[error]      |
[error]      |It would fail on pattern case: _: MethodType, _: PolyType, _: TypeLambda
```

Reply:
Yes, it's unknown that the extractors are irrefutable for the corresponding type `MethodType` or `PolyType `.

Reply:
Should we make `ClassTag.unapply` (and later `TypeTest.unapply`) extractors irrefutable? It feels like they should be.

Reply:
`ClassTag.unapply` is defined as follows:

```Scala
  def unapply(x: Any): Option[T] =
    if (runtimeClass.isInstance(x)) Some(x.asInstanceOf[T])
    else None
```

It's not irrefutable in general. Semantically, we may assume it's irrefutable if we know `x` takes the type `T`.

Reply:
That is the contract of the method. If `x` does not take the type `T` then implementation is broken and we cannot reason about the patterns correctly anyway.

Reply:
Is the following supposed to be irrefutable?

```scala
lambda match
  case lambda: MethodType =>
  case lambda: PolyType =>
  case lambda: TypeLambda =>
```